### ISSUE TITLE
isotope labelled cross-linkers (in mzIdentML)?

### ISSUE THREAD

**Issue by colin-combe** [Rappsilber-Laboratory/xiSPEC_ms_parser#12]
After getting some things wrong on the mzIdentML github project, I now know about isotope labelled cross-linkers.

These have four SII with the same MS:1002511 'value' within the same SIR.

We don't support this aspect of mzIdentMl at the moment.

**Comment by lutzfischer** [comment]
Logically I would assume they have for four SII, in two different SIR.
I assume this is for the xquest way of identification. There are two distinct spectra (therefore two SIR) that make up one identification.

But I find that a bit problematic, as you could also interpret this as a higher order cross-link of some kind.

E.g. if I would write a cross-link between four distinct peptides I would write one SIL with four SII that have the same MS:1002511 'value'.

This way to represent the xquest results sounds like a good way to give everybody a hard time in the future.

I would rather have the heavy and light labeled identifications as two distinct sets of IDs that have potentially another cvparam linking them as a pair.

**Comment by colin-combe** [comment]
from the spec:
> In the case, of “light” and “heavy” cross-linked peptides, there SHOULD be exactly four SpectrumIdentificationItem elements, all sharing the same identifier under “cross-link spectrum
identification item” (light alpha chain, light beta chain, heavy alpha chain, heavy beta chain).

Its not super clear but I think it means within a single SIR.

As far as I can tell, the following is what's been happening with the validator (https://github.com/HUPO-PSI/mzIdentML/issues/105 , https://github.com/HUPO-PSI/mzIdentML/issues/100)...

First it was passing files if there were exactly 2 or 4 occurrences of MS:1002511 with same value in the SIL. Now I think it passes if there are an even number of occurrences of  MS:1002511 with same value in SIL.

I can't test this properly because I can't build the validator.

If you can contribute anything to the HUPO/mzIdentML project to help with this then please do - I made a bit of a mess of it over there.

**Comment by colin-combe** [comment]
yup, from example file https://github.com/HUPO-PSI/mzIdentML/blob/master/examples/1_2examples/crosslinking/OpenxQuest_example.mzid

it looks like 4 within SIR (e.g. search '6054414965487887')

> as you could also interpret this as a higher order cross-link of some kind.

wouldn't they need to be in pairs with same MS:1002511, so you know what ends go together?
